### PR TITLE
feat(jtk): fix sprints list sort order and current board name format

### LINKED ISSUE #286: jtk sprints: fix list sort order and current/list board name format
Parent: #230

### `jtk sprints list --board <id>` (default + extended)
**Current:** sorted oldest-first.
**Expected:** most-recent-first — active sprint first, then future, then closed by recency:
```
ID | STATE | START | END | NAME
125 | active | 2026-04-10 | 2026-04-24 | MON Sprint 70
126 | future | - | - | MON Sprint 71
124 | closed | 2026-03-27 | 2026-04-10 | MON Sprint 69
123 | closed | 2026-03-13 | 2026-03-27 | MON Sprint 68
```

### `jtk sprints list --board <id> --extended`
**Current:** oldest-first; GOAL field contains raw metadata instead of the sprint goal text.
**Expected:**
```
ID | STATE | START | END | COMPLETED | BOARD | GOAL | NAME
125 | active | 2026-04-10 | 2026-04-24 | - | 23 | Ship CapOne a11y fixes | MON Sprint 70
124 | closed | 2026-03-27 | 2026-04-10 | 2026-04-10 | 23 | Complete Q2 integration milestone | MON Sprint 69
```

### `jtk sprints current --board <id>` (default)
**Current:** `Board:` line shows ID only — e.g. `Board: 23`.
**Expected:** `Board: 23 (MON board)` — includes board name resolved from cache.

### `jtk sprints current --board <id> --extended`
**Current:** same missing board name issue.
**Expected:**
```
125  MON Sprint 70
State: active   Start: 2026-04-10T00:00:45Z   End: 2026-04-24T23:30:00Z
Board: 23 (MON board)
Goal: Ship CapOne a11y fixes
Origin Board: 23
```

See #230 for full spec.

### PR DESCRIPTION
**Pull request by rianjs**
## Summary
- Sort `sprints list` output: active → future → closed by recency (CompleteDate → EndDate → StartDate, deterministic ID tie-breaker)
- Fetch all pages before sorting for correct global ordering with client-side `--max`/`--next-page-token` pagination
- Enrich synthetic board in `sprints current` when cache is cold via single `GetBoard` API call (graceful fallback)

Closes #286

## Test plan
- [x] `TestSortSprintsForDisplay` — mixed states, dates, nil dates, deterministic tie-breaking
- [x] `TestSortSprintsForDisplay_ClosedByCompleteDate` — closed sort by CompleteDate
- [x] `TestSortSprintsForDisplay_DeterministicTieBreaker` — ID-based tie-breaking
- [x] `TestRunList_SortOrder` — table output in state-priority order
- [x] `TestRunList_SortOrder_IDOnly` — --id respects sort order
- [x] `TestRunList_SortOrder_JSON` — JSON respects sort order
- [x] `TestRunList_Extended_GoalColumn` — goal text rendered correctly after sort
- [x] `TestRunList_ClientSidePagination` — --max and --next-page-token window
- [x] `TestRunCurrent_BoardEnrichment` — synthetic board gets name from API
- [x] `TestRunCurrent_BoardEnrichmentFailsGracefully` — 404 → shows ID only
- [x] All 38 sprint command tests pass
- [x] All 225 presenter tests pass
- [x] `make build`, `make test`, `make lint` clean (jtk)

### CODE REVIEW

**Comment by rianjs** [comment]
**Blocker**
- None.

**Major**
- `tools/jtk/internal/cmd/sprints/sprints.go` / `fetchAllSprints`: the loop can silently return an incomplete dataset. `len(result.Values) == 0` with `IsLast=false` should remain an error, not a successful break, and the `len(all) > 10000` guard should either be removed or return an explicit error. Silent truncation violates the fetch-all-before-sort requirement.
- `tools/jtk/internal/cmd/sprints/sprints.go` / client-side windowing: `maxResults <= 0` is now unsafe. A negative `--max` can produce an invalid slice range, and `--max 0` now reports “No sprints found” even when data exists. Validate `--max > 0` or normalize before slicing.

**Minor**
- `tools/jtk/internal/cmd/sprints/sprints_test.go`: board enrichment tests still call `runCurrent` directly with a synthetic board. The plan called for Cobra-level coverage of `sprints current --board 23`, which is the resolver numeric-pass-through path that caused the bug.

**Nit**
- None.

**Comment by rianjs** [comment]
## TDD Assessment — PR #296 (sprint sort + board enrichment)

### What changed
1. `fetchAllSprints` — new internal fetch-all loop replacing single-page API call
2. `SortSprintsForDisplay` — new sort function in `present/sprint.go`
3. `runList` — now does client-side pagination over the sorted slice
4. `runCurrent` — board name enrichment when board is synthetic (no name)

---

### Critical paths

- **`fetchAllSprints` error propagation** — the `return nil, err` branch (API returns non-200 mid-pagination) has no dedicated test. It's an untested return path in a loop.
- **`fetchAllSprints` 10 000-sprint safety cap** — the `len(all) > 10000` guard exits silently with a truncated slice. No test verifies the cap fires or that the caller is unaware of truncation. This is a silent data loss path.
- **`fetchAllSprints` with `IsLast=false` and empty page** — the old code explicitly errored on this contract violation (`unexpected paginated response: IsLast=false with empty values`). The new loop just breaks silently. The removal of that guard is unacknowledged and untested.

### Edge cases

- **`closedLess` EndDate/StartDate fallback** — `TestSortSprintsForDisplay_ClosedByCompleteDate` only tests the `CompleteDate` branch. The `EndDate` and `StartDate` fallback branches in `closedLess` are not directly exercised.
- **Multiple active sprints** — the sort places all actives first, tie-broken by StartDate-desc then ID. No test has two active sprints with different StartDates to verify that ordering.
- **Multiple future sprints** — same gap; only one future sprint appears in every sort test.
- **`startAt > total` clamping** — the guard `if startAt > total { startAt = total }` produces an empty page. No test passes an out-of-bounds token to confirm the empty-page path instead of a panic.
- **`maxResults == 0`** — would produce a zero-length slice window; not tested.

### Business logic gaps

- **Board enrichment skips JSON/ID-only paths** — enrichment only matters for table display (`Board: 23 (MON board)`). Tests confirm the table path but there is no assertion that JSON output also includes the enriched name (or that it intentionally does not).
- **Board enrichment with wrong ID** — the guard `enriched.ID == board.ID` prevents a mismatched response from being used, but there is no test that sends a board response with a mismatched ID to verify the fallback.
- **State filter passthrough to `fetchAllSprints`** — `runList` passes the `state` flag to `fetchAllSprints`, but all new tests use `state=""`. No test verifies that `state=active` is forwarded to the API query parameter on each page of a multi-page fetch.

### Reviewer flags

- The `IsLast=false + len(Values)==0` contract guard was deliberately removed — this should be called out in the PR description as an intentional behavior change, with rationale. Its absence is easy to miss.
- `fetchPageSize = 50` is a package-level constant with no test asserting it is sent correctly to the API (the existing pagination server test uses page-size 2, not 50).
- The 10 000-sprint cap is undocumented in user-facing help or comments beyond the code itself.

---

### Verdict

**Coverage is solid for the happy-path and the headline features** (sort order across all three output formats, client-side pagination, board enrichment success/failure). The gaps are concentrated in:
1. Error propagation through `fetchAllSprints` (untested error return)
2. The silent removal of the `IsLast=false + empty-values` guard (behavioral regression risk)
3. The 10 000-sprint truncation cap (silent data loss, zero tests)

These are worth addressing before merge, especially #2 and #3.

**Comment by rianjs** [comment]
**Blocker**
- None.

**Major**
- None.

**Minor**
- None.

**Nit**
- None.

File: tools/jtk/internal/cmd/sprints/sprints.go [hunk 1]
```diff
@@ -252,6 +286,13 @@ func runCurrent(ctx context.Context, opts *root.Options, client *api.Client, boa
 		return v.RenderArtifact(jtkartifact.ProjectSprint(sprint, opts.ArtifactMode()))
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/sprints/sprints.go:286]: 🟡 **Medium** (harness-engineering:harness-self-documenting-code-reviewer): Board enrichment is placed after the JSON early-return, so `sprints current --output json` never enriches the synthetic board name. A synthetic board (cache miss or numeric --board flag) always emits an empty `name` field in JSON output, which is inconsistent with the PR's stated goal of enriching the board name when the cache is cold.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/sprints/sprints.go [hunk 2]
```diff
@@ -144,44 +144,78 @@ func runList(ctx context.Context, opts *root.Options, client *api.Client, boardI
...
 		return err
 	}
 
-	hasMore := !result.IsLast
-	if hasMore && len(result.Values) == 0 {
-		return fmt.Errorf("unexpected paginated response: IsLast=false with empty values (startAt=%d)", startAt)
+	jtkpresent.SortSprintsForDisplay(allSprints)
+
+	// Client-side pagination window over the sorted slice.
+	if maxResults <= 0 {
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/sprints/sprints.go:155]: 🔵 **Low** (harness-engineering:harness-self-documenting-code-reviewer): Every `sprints list` invocation now fetches the entire sprint history before applying --max, making the call O(N/50) API requests instead of the previous O(1). A board with 200+ closed sprints issues 4+ sequential API calls even when the user requests --max 1. This is an intentional tradeoff for correct global sort order, but there is no user-visible warning or documentation of the behavioral change.

*Reply to this thread when addressed.*

File: tools/jtk/internal/present/sprint.go [hunk 3]
```diff
@@ -2,13 +2,82 @@ package present
...
+		return 1
+	default:
+		return 2
+	}
+}
+
+func closedLess(a, b api.Sprint) bool {
+	if cmp := compareDatesDesc(a.CompleteDate, b.CompleteDate); cmp != 0 {
+		return cmp < 0
+	}
```
Comment (monit-reviewer) [inline tools/jtk/internal/present/sprint.go:48]: 🔵 **Low** (harness-engineering:harness-knowledge-reviewer): Future sprints are sorted by StartDate descending (most distant future first), reusing the active-sprint sort logic. For future sprints the more natural UX is ascending — nearest upcoming sprint first. The PR description does not justify applying descending order to future sprints.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/sprints/sprints_test.go [hunk 4]
```diff
@@ -843,6 +861,366 @@ func TestAddCmd_NumericSprintPassThrough(t *testing.T) {
...
+
+	output = stdout.String()
+	testutil.Equal(t, output, "1\n")
+}
+
+func TestRunCurrent_BoardEnrichment(t *testing.T) {
+	t.Parallel()
+
+	seedBoardsAndSprints(t, nil, nil) // empty cache — board resolves synthetic
+
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/sprints/sprints_test.go:1085]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): TestCurrentCmd_BoardEnrichmentCobraLevel is missing t.Parallel() while TestRunCurrent_BoardEnrichment — which also calls seedBoardsAndSprints — is marked parallel. If seedBoardsAndSprints touches shared cache state, the asymmetry makes this test vulnerable to data races under -count>1 or certain test orderings.

*Reply to this thread when addressed.*

File: tools/jtk/internal/present/sprint.go [hunk 5]
```diff
@@ -2,13 +2,82 @@ package present
...
 	"github.com/open-cli-collective/jira-ticket-cli/internal/present/projection"
 )
 
+// SortSprintsForDisplay sorts sprints for display: active first, then future,
+// then closed by recency. Within closed, sort by CompleteDate → EndDate →
+// StartDate descending. Deterministic tie-breaker: ID descending.
+func SortSprintsForDisplay(sprints []api.Sprint) {
+	sort.SliceStable(sprints, func(i, j int) bool {
+		pi, pj := statePriority(sprints[i].State), statePriority(sprints[j].State)
+		if pi != pj {
```
Comment (monit-reviewer) [inline tools/jtk/internal/present/sprint.go:20]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): statePriority returns the closed-sprint priority for any unrecognized sprint state. If Jira introduces a new state (e.g., 'archived'), it silently sorts alongside closed sprints and applies closed-sort date logic, producing confusing output with no warning.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/sprints/sprints.go [hunk 6]
```diff
@@ -144,44 +144,78 @@ func runList(ctx context.Context, opts *root.Options, client *api.Client, boardI
...
+		maxResults = 50
+	}
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/sprints/sprints.go:157]: 🔵 **Low** (harness-engineering:harness-knowledge-reviewer): When --next-page-token (startAt) exceeds the total sprint count, the client-side windowing silently returns an empty page with no error. Stale or invalid tokens are indistinguishable from a legitimately empty last page, making pagination bugs harder to diagnose.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/sprints/sprints_test.go [hunk 7]
```diff
@@ -234,24 +235,41 @@ func TestRunList_InvalidNextPageToken(t *testing.T) {
...
+		}
+		page := allSprints[startAt:end]
 		_ = json.NewEncoder(w).Encode(api.SprintsResponse{
-			Values: []api.Sprint{{ID: 10, Name: "S1", State: "active"}},
-			IsLast: false,
+			Values: page,
+			IsLast: end >= len(allSprints),
 		})
 	}))
 	defer server.Close()
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/sprints/sprints_test.go:260]: 🔵 **Low** (harness-engineering:harness-self-documenting-code-reviewer): TestRunList_Pagination was refactored to IDOnly mode, so no test now verifies that client-side pagination emits the 'More results available' token in table (non-ID-only, non-JSON) output mode.

*Reply to this thread when addressed.*

**Review by monit-reviewer — DISMISSED** [review]
## Automated PR Review



**Reviewed commit:** `8b13265`

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-architecture-reviewer | 2 |
| harness-engineering:harness-knowledge-reviewer | 2 |
| harness-engineering:harness-self-documenting-code-reviewer | 3 |

<details>
<summary><strong>harness-engineering:harness-architecture-reviewer (2 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cmd/sprints/sprints_test.go:1055`
> TestCurrentCmd_BoardEnrichmentCobraLevel is missing t.Parallel() while TestRunCurrent_BoardEnrichment — which also calls seedBoardsAndSprints — is marked parallel. If seedBoardsAndSprints touches shared cache state, the asymmetry makes this test vulnerable to data races under -count>1 or certain test orderings.

**💡 Suggestion** - `tools/jtk/internal/present/sprint.go:20`
> statePriority returns the closed-sprint priority for any unrecognized sprint state. If Jira introduces a new state (e.g., 'archived'), it silently sorts alongside closed sprints and applies closed-sort date logic, producing confusing output with no warning.

</details>

<details>
<summary><strong>harness-engineering:harness-knowledge-reviewer (2 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/present/sprint.go:44`
> Future sprints are sorted by StartDate descending (most distant future first), reusing the active-sprint sort logic. For future sprints the more natural UX is ascending — nearest upcoming sprint first. The PR description does not justify applying descending order to future sprints.

**💡 Suggestion** - `tools/jtk/internal/cmd/sprints/sprints.go:157`
> When --next-page-token (startAt) exceeds the total sprint count, the client-side windowing silently returns an empty page with no error. Stale or invalid tokens are indistinguishable from a legitimately empty last page, making pagination bugs harder to diagnose.

</details>

<details>
<summary><strong>harness-engineering:harness-self-documenting-code-reviewer (3 findings)</strong></summary>

**⚠️ Should Fix** - `tools/jtk/internal/cmd/sprints/sprints.go:286`
> Board enrichment is placed after the JSON early-return, so `sprints current --output json` never enriches the synthetic board name. A synthetic board (cache miss or numeric --board flag) always emits an empty `name` field in JSON output, which is inconsistent with the PR's stated goal of enriching the board name when the cache is cold.

**💡 Suggestion** - `tools/jtk/internal/cmd/sprints/sprints.go:155`
> Every `sprints list` invocation now fetches the entire sprint history before applying --max, making the call O(N/50) API requests instead of the previous O(1). A board with 200+ closed sprints issues 4+ sequential API calls even when the user requests --max 1. This is an intentional tradeoff for correct global sort order, but there is no user-visible warning or documentation of the behavioral change.

**💡 Suggestion** - `tools/jtk/internal/cmd/sprints/sprints_test.go:260`
> TestRunList_Pagination was refactored to IDOnly mode, so no test now verifies that client-side pagination emits the 'More results available' token in table (non-ID-only, non-JSON) output mode.

</details>

*2 info-level observations excluded. Run with `--verbose` to include.*

*3 PR discussion threads considered.*

---
<details>
<summary>Completed in 4m 01s | $1.07</summary>

| Field | Value |
|---|---|
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 4m 01s wall · 9m 51s compute (Reviewers: 2m 05s · Synthesis: 1m 53s) |
| Cost | $1.07 |
| Tokens | 345.3k in / 35.3k out |
| Turns | 17 |

</details>

File: tools/jtk/internal/cmd/sprints/sprints.go (hunk 1 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/sprints/sprints.go:286]: The Sprint struct doesn't carry the board name — it's only available in the presentation context. JSON output emits the Sprint artifact directly, so there's no board name field to enrich.

File: tools/jtk/internal/cmd/sprints/sprints.go (hunk 2 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/sprints/sprints.go:155]: Acknowledged — intentional tradeoff for correct global sort order. Boards typically have <200 sprints (4 API calls). The --state filter still reduces server-side volume.

File: tools/jtk/internal/present/sprint.go (hunk 3 above)
Comment (rianjs) [reply tools/jtk/internal/present/sprint.go:48]: Fixed — future sprints now sort ascending (nearest upcoming first), with nil StartDate sorting last.

File: tools/jtk/internal/cmd/sprints/sprints_test.go (hunk 4 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/sprints/sprints_test.go:1085]: Fixed — added t.Parallel().

File: tools/jtk/internal/present/sprint.go (hunk 5 above)
Comment (rianjs) [reply tools/jtk/internal/present/sprint.go:20]: The default case is intentional — unknown states sort with closed as the safe default. Adding warnings for hypothetical future states would be speculative.

File: tools/jtk/internal/cmd/sprints/sprints.go (hunk 6 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/sprints/sprints.go:157]: Consistent with all other paginated commands — stale tokens return empty results.

File: tools/jtk/internal/cmd/sprints/sprints_test.go (hunk 7 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/sprints/sprints_test.go:260]: Fixed — added TestRunList_ClientSidePagination_Table.

File: tools/jtk/internal/cmd/sprints/sprints.go (hunk 1 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cmd/sprints/sprints.go:286]: **Settled**: monit-reviewer flagged that board enrichment is skipped for JSON output, leaving the name field empty for synthetic boards. rianjs explained that the Sprint struct doesn't carry the board name — it's only available in presentation context — so there's no field to enrich in JSON output.

File: tools/jtk/internal/cmd/sprints/sprints.go (hunk 2 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cmd/sprints/sprints.go:155]: **Settled**: monit-reviewer noted that fetching all sprint history before applying --max is now O(N/50) API requests with no user-visible warning. rianjs acknowledged this as an intentional tradeoff for correct global sort order, noting boards typically have fewer than 200 sprints and the --state filter still reduces server-side volume.

File: tools/jtk/internal/present/sprint.go (hunk 3 above)
Comment (monit-reviewer) [reply tools/jtk/internal/present/sprint.go:48]: **Settled**: monit-reviewer flagged that future sprints sorted descending (most distant first) is unintuitive UX. rianjs fixed it — future sprints now sort ascending with nearest upcoming first and nil StartDate sorting last.

File: tools/jtk/internal/cmd/sprints/sprints_test.go (hunk 4 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cmd/sprints/sprints_test.go:1085]: **Settled**: monit-reviewer noted that TestCurrentCmd_BoardEnrichmentCobraLevel was missing t.Parallel() while a related test was marked parallel, creating a potential data race. rianjs fixed it by adding t.Parallel().

File: tools/jtk/internal/present/sprint.go (hunk 5 above)
Comment (monit-reviewer) [reply tools/jtk/internal/present/sprint.go:20]: **Settled**: monit-reviewer flagged that statePriority silently sorts unrecognized sprint states alongside closed sprints with no warning. rianjs explained this is intentional — unknown states defaulting to closed is the safe fallback, and adding warnings for hypothetical future states would be speculative.

File: tools/jtk/internal/cmd/sprints/sprints.go (hunk 6 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cmd/sprints/sprints.go:157]: **Settled**: monit-reviewer noted that stale or invalid pagination tokens silently return empty results, making pagination bugs hard to diagnose. rianjs confirmed this is consistent behavior with all other paginated commands in the tool.

File: tools/jtk/internal/cmd/sprints/sprints_test.go (hunk 7 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cmd/sprints/sprints_test.go:260]: **Settled**: monit-reviewer flagged that refactoring TestRunList_Pagination to IDOnly mode left no test verifying the 'More results available' token in table output mode. rianjs fixed it by adding TestRunList_ClientSidePagination_Table.

File: tools/jtk/internal/cmd/sprints/sprints_test.go [hunk 8]
```diff
@@ -843,6 +861,397 @@ func TestAddCmd_NumericSprintPassThrough(t *testing.T) {
...
+	client, err := api.New(api.ClientConfig{URL: server.URL, Email: "test@test.com", APIToken: "token"})
+	testutil.RequireNoError(t, err)
+
+	var stdout bytes.Buffer
+	opts := &root.Options{Output: "json", Stdout: &stdout, Stderr: &bytes.Buffer{}}
+	opts.SetAPIClient(client)
+
+	err = runList(context.Background(), opts, client, 123, "", 50, "", "")
+	testutil.RequireNoError(t, err)
+
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/sprints/sprints_test.go:960]: 🔵 **Low** (harness-engineering:harness-self-documenting-code-reviewer): TestRunList_SortOrder_JSON verifies sort order by comparing raw string positions of 'Active', 'Future', and 'Closed' in the JSON output via strings.Index. This is fragile: if any other field value or key name in the serialized output contains those words at an earlier position, the assertion silently gives the wrong result. Unmarshaling the JSON array and asserting on element order by ID would be unambiguous.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/sprints/sprints_test.go [hunk 9]
```diff
@@ -843,6 +861,397 @@ func TestAddCmd_NumericSprintPassThrough(t *testing.T) {
...
+	// Page 2: startAt=2, max=2
+	stdout.Reset()
+	err = runList(context.Background(), opts, client, 123, "", 2, "2", "")
+	testutil.RequireNoError(t, err)
+
+	output = stdout.String()
+	testutil.Equal(t, output, "1\n")
+}
+
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/sprints/sprints_test.go:1050]: 🔵 **Low** (harness-engineering:harness-enforcement-reviewer): TestRunCurrent_BoardEnrichmentFailsGracefully does not call seedBoardsAndSprints, unlike its sibling TestRunCurrent_BoardEnrichment. If a prior parallel test populates the shared board cache, the failing-gracefully test may exercise a different code path than intended (cache hit vs. synthetic board). The setup inconsistency should be explained with a comment or aligned with the other enrichment test.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/sprints/sprints.go [hunk 10]
```diff
@@ -144,44 +144,78 @@ func runList(ctx context.Context, opts *root.Options, client *api.Client, boardI
...
 	}
 
 	if idOnly {
-		ids := make([]string, len(result.Values))
-		for i, s := range result.Values {
+		ids := make([]string, len(page))
+		for i, s := range page {
 			ids[i] = strconv.Itoa(s.ID)
 		}
 		return jtkpresent.EmitIDsWithPaginationToken(opts, ids, hasMore, nextToken)
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/sprints/sprints.go:178]: 🔵 **Low** (harness-engineering:harness-knowledge-reviewer): fetchAllSprints still has no explicit upper-bound guard on total loop iterations. All four specialized reviewers flag this on the updated code. A misbehaving server that returns IsLast=false indefinitely with non-empty pages causes unbounded API calls and memory growth. A max-pages or max-items guard with an explicit error (not silent truncation) would make the failure mode clear.

*Reply to this thread when addressed.*

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `0f4850a` | **Previous:** `8b13265` (incremental)

> **Approved** with 3 non-blocking suggestions below. Address at your discretion.

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-enforcement-reviewer | 1 |
| harness-engineering:harness-knowledge-reviewer | 1 |
| harness-engineering:harness-self-documenting-code-reviewer | 1 |

<details>
<summary><strong>harness-engineering:harness-enforcement-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cmd/sprints/sprints_test.go:1050`
> TestRunCurrent_BoardEnrichmentFailsGracefully does not call seedBoardsAndSprints, unlike its sibling TestRunCurrent_BoardEnrichment. If a prior parallel test populates the shared board cache, the failing-gracefully test may exercise a different code path than intended (cache hit vs. synthetic board). The setup inconsistency should be explained with a comment or aligned with the other enrichment test.

</details>

<details>
<summary><strong>harness-engineering:harness-knowledge-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cmd/sprints/sprints.go:178`
> fetchAllSprints still has no explicit upper-bound guard on total loop iterations. All four specialized reviewers flag this on the updated code. A misbehaving server that returns IsLast=false indefinitely with non-empty pages causes unbounded API calls and memory growth. A max-pages or max-items guard with an explicit error (not silent truncation) would make the failure mode clear.

</details>

<details>
<summary><strong>harness-engineering:harness-self-documenting-code-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cmd/sprints/sprints_test.go:960`
> TestRunList_SortOrder_JSON verifies sort order by comparing raw string positions of 'Active', 'Future', and 'Closed' in the JSON output via strings.Index. This is fragile: if any other field value or key name in the serialized output contains those words at an earlier position, the assertion silently gives the wrong result. Unmarshaling the JSON array and asserting on element order by ID would be unambiguous.

</details>

*2 info-level observations excluded. Run with `--verbose` to include.*

*10 PR discussion threads considered.*

---
<details>
<summary>Completed in 3m 45s | $1.07</summary>

| Field | Value |
|---|---|
| Mode | Re-review · Cycle 2 · Session resumed |
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 3m 45s wall · 8m 43s compute (Reviewers: 1m 47s · Synthesis: 1m 34s) |
| Cost | $1.07 |
| Tokens | 272.9k in / 30.5k out |
| Turns | 12 |

</details>
